pulsar-client/src/main/java/org/apache/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -2294,17 +2295,16 @@ private CompletableFuture<Producer<byte[]>> initDeadLetterProducerIfNeeded() {
                 p = deadLetterProducer;
                 if (p == null || p.isCompletedExceptionally()) {
                     p = createProducerWithBackOff(() -> {
-                        CompletableFuture<Producer<byte[]>> newProducer =
-                                ((ProducerBuilderImpl<byte[]>) client.newProducer(Schema.AUTO_PRODUCE_BYTES(schema)))
-                                        .initialSubscriptionName(this.deadLetterPolicy.getInitialSubscriptionName())
-                                        .topic(this.deadLetterPolicy.getDeadLetterTopic())
-                                        .producerName(
-                                                String.format("%s-%s-%s-%s-DLQ", this.topicName, this.subscription,
-                                                        this.consumerName, RandomStringUtils.randomAlphanumeric(5)))
-                                        .blockIfQueueFull(false)
-                                        .enableBatching(false)
-                                        .enableChunking(true)
```

Comment:
keep these as explained in the previous comment

Reply:
There was a slight misunderstanding regarding the `DEFAULT_PRODUCER_BUILDER_CUSTOMIZER`. Remove it completely, it's not needed when you revert the change to remove these lines above.

If the user would like to change the above settings, they could do that in the customizer. It also is important for the "customizer" abstraction that unless you do any changes in the customizer, the defaults will be use. The customizer customizes the producer builder after the runtime has configured the defaults. Makes sense?

Reply:
Maybe you can suggest any code, or it is difficult to understand the subtle difference between current implementation and your expectation.

Reply:
@thetumbled I pushed directly to this PR, f353e02. There was another issue that the ProducerBuilderContextImpl was in the wrong package. This is a class that should be kept private in the same package as ConsumerImpl. It could even be a private inner class in ConsumerImpl, there's no value in reusing it.